File: angry-professor.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        int t, k, n, present, tmp;
        Scanner scan = new Scanner(System.in);
        t = Integer.parseInt(scan.nextLine());
        for(int i=0; i<t;i++) {
            n = scan.nextInt();
            k = scan.nextInt();
            present = 0;
            for(int j=0;j<n;j++){
                tmp = scan.nextInt();
                if(tmp<=0) present++;
            }
            if(present<k) System.out.println("YES");
            else System.out.println("NO");
        }
        
    }
}

File: Algorithms Implementation/Easy/angry-professor.js
var n = 4;
var k = 2;
var a = "-1 -3 4 2".split(" ");
var present = 0;
for(var i=0;i<n;i++) {
	if(a[i]<=0) present++;
}
if(present<k) console.log("YES");
else console.log("NO");